### PR TITLE
nordpass: init at 5.19.3

### PR DESCRIPTION
**Pull request by coreyoconnor**
Built from the snap distribution.

## Description of changes

Adds nordpass password manager gui application. This builds from the snap distribution. See #230185 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/no/nordpass/package.nix [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+, glib, pango, cairo, atk, gdk-pixbuf, gtk3, cups, nspr, nss_latest, libpng, libnotify
+, libgcrypt, systemd, fontconfig, dbus, expat, curlWithGnuTls, zlib, gnome
+, at-spi2-atk, at-spi2-core, libdrm, mesa, libxkbcommon
+, harfbuzz, libsecret, buildFHSEnv
+}:
+
+let
+  version = "5.14.23";
+
+  deps = pkgs: [
```
Comment (Aleksanaa) [inline pkgs/by-name/no/nordpass/package.nix:11]:
```suggestion
  deps = [
```

File: pkgs/by-name/no/nordpass/package.nix [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+buildFHSEnv {
+  name = "nordpass";
+  targetPkgs = pkgs: (deps pkgs) ++ [ thisPackage ];
+  runScript = "nordpass";
+
+  extraInstallCommands = ''
+    mkdir -p "$out/share"
+    cp -r ${thisPackage}/share/* "$out/share/"
+  '';
```
Comment (Aleksanaa) [inline pkgs/by-name/no/nordpass/package.nix:127]: Maybe `inherit (thisPackage) meta;`?

File: pkgs/by-name/no/nordpass/package.nix [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        runHook postInstall
+      '';
+
+    meta = with lib; {
+      homepage = "https://nordpass.com/";
+      description = "NordPass application re-packaged from the snap distribution";
+      license = licenses.unfree;
+      mainProgram = "nordpass";
+      maintainers = with maintainers; [ coconnor ];
+      platforms = platforms.linux;
```
Comment (Aleksanaa) [inline pkgs/by-name/no/nordpass/package.nix:114]:
```suggestion
      platforms = platforms.linux;
      sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

Comment (Aleksanaa) [inline pkgs/by-name/no/nordpass/package.nix:114]:
```suggestion
      platforms = [ "x86_64-linux" ];
```

File: pkgs/by-name/no/nordpass/package.nix [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+  thisPackage = stdenv.mkDerivation {
+    pname = "nordpass";
+
+    inherit version;
+
+    # determine from
+    # curl -H 'Snap-Device-Series: 16' http://api.snapcraft.io/v2/snaps/info/nordpass
+    src = fetchurl {
+      url = "https://api.snapcraft.io/api/v1/snaps/download/00CQ2MvSr0Ex7zwdGhCYTa0ZLMw3H6hf_178.snap";
```
Comment (Aleksanaa) [inline pkgs/by-name/no/nordpass/package.nix:65]: Is this "00CQ2MvSr0Ex7zwdGhCYTa0ZLMw3H6hf_178" reproducible?

File: pkgs/by-name/no/nordpass/package.nix [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+        sed -i -e "s#^Icon=.*\$#Icon=$out/share/icons/hicolor/512x512/apps/nordpass.png#" \
+          "$out/share/applications/nordpass.desktop"
+
+        runHook postInstall
+      '';
+
+    meta = with lib; {
+      homepage = "https://nordpass.com/";
+      description = "NordPass application re-packaged from the snap distribution";
```
Comment (Aleksanaa) [inline pkgs/by-name/no/nordpass/package.nix:110]:
```suggestion
      description = "A secure and simple password manager for a stress-free online experience";
```

File: pkgs/by-name/no/nordpass/package.nix (hunk 2 above)
Comment (coreyoconnor) [reply pkgs/by-name/no/nordpass/package.nix:127]: I was wondering if this should have meta attributes! Thanks. will do.

File: pkgs/by-name/no/nordpass/package.nix (hunk 4 above)
Comment (coreyoconnor) [reply pkgs/by-name/no/nordpass/package.nix:65]: Not sure by "reproducible". As far as I can tell that is a content based hash. That URL will be the same for this version of the application. Couldn't find a spec on this tho.

Comment (coreyoconnor) [reply pkgs/by-name/no/nordpass/package.nix:65]: This is not a content based ID.
This is a unique Id assigned to the "snap". The application ID. This is a fixed Id for the application. The `178` suffix is the version of the app. So, Technically producible as "snap-id" + "_" + "version number"

File: pkgs/by-name/no/nordpass/package.nix (hunk 2 above)
Comment (coreyoconnor) [reply pkgs/by-name/no/nordpass/package.nix:131]: added

File: pkgs/by-name/no/nordpass/package.nix (hunk 4 above)
Comment (coreyoconnor) [reply pkgs/by-name/no/nordpass/package.nix]: ok. I broke this URL up into components. Seems like the `snapBaseUrl` should go elsewhere but *shrug*

**Comment by coreyoconnor** [comment]
updated to 5.18.7

**Comment by coreyoconnor** [comment]
Closing. I'm no longer packaging apps in `nixpkgs` unless they require specific hardware options. Creating a nix flake for this.

**Comment by Aleksanaa** [comment]
I'm sorry for your bad experience. Our mailbox are always filled with tons of notifications, and reviewed queue are always 200+. While you are waiting for the feedback, you can ask for another review at https://discourse.nixos.org/t/prs-ready-for-review/3032/4531, https://discourse.nixos.org/t/prs-already-reviewed/2617/1656 and https://matrix.to/#/#review-requests:nixos.org. We are still trying to improve our workflow, but there are still dead ends.

**Comment by coreyoconnor** [comment]
Not a bad experience!

I've had such a good experience with flakes that i really think they are
the best way for these kinds of apps.

**Comment by coreyoconnor** [comment]
Specifically: This is a third party binary with no hardware specific requirements. That seems weird to put in nixpkgs. If people disagree then I can re-open this PR.

**Comment by Kek5chen** [comment]
I do disagree. It would be nice to have nordpass and nordvpn in nixpkgs. Currently NordVPN is only found on NUR, which i see more as a staging environment until packages go live in nixpkgs. Steam is also a third party binary, but people depend on it, and while unfree, the nord packages would be nice to have.

Though I am a bit new to the Nix scene, so would you mind hinting me towards where I can get nordpass on the alternative way through the flake? Where do I find it? I don't see any contributions on your profile regarding that, or I'm just blind >-> .

Thank you!

**Comment by coreyoconnor** [comment]
I forgot to mention: I had the impression the nordpass application was being deprecated. The help page definitely treats the desktop app as a secondary thing ( https://support.nordpass.com/hc/en-us/articles/360006638797-Using-NordPass-on-desktop ) But definitely seems like the application is being supported once you click through the link? I can't find the link where I got that impression from now tho.

I'm unfamiliar with NordVPN.  Does that have a similar app?

**Comment by Kek5chen** [comment]
The nordvpn app is actually already on the NUR. But it would be nice to have it on nixpkgs. Anyways. The NordPass app definitely is a secondary thing but it's definitely not being deprecated. Nord provides a bunch of services together.

**Comment by coreyoconnor** [comment]
Understood! Re-opened at https://github.com/NixOS/nixpkgs/pull/353552

**Pull request closed without merging**